lrlex/src/main.rs

```diff
@@ -26,14 +26,18 @@ fn usage(prog: &str, msg: &str) {
 }
 
 fn read_file(path: &str) -> String {
```

Comment:
It might be worth noting here, that the `read_file` functions between lrlex and nimbleparse,
seem to have diverged in their argument types:

`lrlex` takes an `&str`, which `nimbleparse` takes an `AsRef<Path>`, and the difference effects these two implementations because there is no `Path*` implementation for `PartialEq<&str>`.

This explains why in lrlex I implemented the logic *inside* `read_file`, while in nimbleparse the logic ended up outside.

Reply:
I'd be happy to see the two homogenised in a future PR!